lib/cucumber/formatter/legacy_api/adapter.rb

```diff
@@ -54,7 +54,7 @@ def printer
         end
 
         def record_test_case_result(test_case, result)
-          scenario = LegacyResultBuilder.new(result).scenario(test_case.name, test_case.location)
+          scenario = LegacyResultBuilder.new(result).scenario("#{test_case.keyword}: #{test_case.name}", test_case.location)
```

Comment:
To be honest, I don't care too much about the code here in the legacy API adapter. We're planning to pull it out into a separate gem as soon as Cucumber 2.0 is released - I don't want to have to maintain this API anymore. So the code doesn't need to be perfect. There are lots worse problems in this file already! Just ask @brasmusson ;)